src/connection/sqlJob.ts

```diff
@@ -181,19 +181,19 @@ export class SQLJob {
 
     const connectResult: ConnectionResult = JSON.parse(result);
 
-    if (connectResult.success !== true) {
+    if (connectResult.success === true) {
+      this.status = JobStatus.Ready;
+    } else {
       this.dispose();
       this.status = JobStatus.NotStarted;
       throw new Error(connectResult.error || `Failed to connect to server.`);
     }
 
-    if (this.status === JobStatus.Ended) {
+    if (this.status !== JobStatus.Ready) {
```

Comment:
Is this even necessary?  It doesn't seem like this could ever happen.  In the `if-else` above, status is either set to `JobStatus.Ready`, or we dispose and throw an `Error`.

Reply:
True story!